src/Features/CSharp/Portable/BraceCompletion/CurlyBraceCompletionService.cs

```diff
@@ -0,0 +1,511 @@
+                        {
+                            var overlappingAmount = formattingEdit.Span.Start - newLineTextStart;
+                            var adjustedFormatChange = new TextChange(new TextSpan(formattingEdit.Span.Start, formattingEdit.Span.Length - overlappingAmount), formattingEditText);
+
+                            // Remove the overlap at the beginning of the new line text.
+                            newLineTextAfterMerge = newLineTextAfterMerge.Remove(0, overlappingAmount);
+                            overallChanges.Add(adjustedFormatChange);
+                        }
+                        // Case 3: The beginning of the formatting change span overlaps with the end of the new line text.
+                        // Remove the overlapping text from the new line edit (the text is included in the formatting change).
+                        // Modify the formatting change span to be everything after the new line edit end (and make the endpoint relative to the original text).
+                        else
+                        {
+                            var overlappingAmount = newLineTextEnd - formattingEdit.Span.Start;
+                            var adjustedFormatChange = new TextChange(new TextSpan(newLineEdit.Span.End, formattingEdit.Span.Length - overlappingAmount), formattingEditText);
+
+                            // Remove the overlap at the end of the new line text.
+                            newLineTextAfterMerge = newLineTextAfterMerge.Substring(0, newLineTextAfterMerge.Length - overlappingAmount);
+                            overallChanges.Add(adjustedFormatChange);
+                        }
+                    }
+                }
+
+                if (newLineTextAfterMerge != string.Empty)
+                {
+                    // Ensure the new line change comes before formatting changes in case of ties.
+                    overallChanges.Insert(0, new TextChange(newLineEdit.Span, newLineTextAfterMerge));
+                }
+
+                return overallChanges.ToImmutable();
+            }
+        }
+
+        public override async Task<bool> CanProvideBraceCompletionAsync(char brace, int openingPosition, Document document, CancellationToken cancellationToken)
+        {
+            // Only potentially valid for curly brace completion if not in an interpolation brace completion context.
+            if (OpeningBrace == brace && await InterpolationBraceCompletionService.IsPositionInInterpolationContextAsync(document, openingPosition, cancellationToken).ConfigureAwait(false))
+            {
+                return false;
+            }
+
+            return await base.CanProvideBraceCompletionAsync(brace, openingPosition, document, cancellationToken).ConfigureAwait(false);
+        }
+
+        protected override bool IsValidOpeningBraceToken(SyntaxToken token) => token.IsKind(SyntaxKind.OpenBraceToken) && !token.Parent.IsKind(SyntaxKind.Interpolation);
```

Comment:
```suggestion
        protected override bool IsValidOpeningBraceToken(SyntaxToken token)
            => token.IsKind(SyntaxKind.OpenBraceToken) && !token.Parent.IsKind(SyntaxKind.Interpolation);
```

so, for *methods*, we prefer to always wrap and indent for expr-bodies.  ihaven't minded up till now because teh methods were trivial.  but if there's going to be complexity, better to just wrap/indent all of them.